lib/facility_factory.rb

```diff
@@ -0,0 +1,42 @@
+class FacilityFactory
+  attr_reader :facilities
+
+  def initialize
+    @facilities = []
+  end
+
+  def create_facilities(items)
+    items.map do |item|
+      if item[:state].include?("CO")
+        new_facility = Facility.new({
+          name: item[:dmv_office], 
+          address: "#{item[:address_li]} #{item[:address__1]} 
+          #{item[:city]} #{item[:state]} #{item[:zip]}", 
+          phone: item[:phone], 
+          hours: item[:hours]})
+        new_facility.add_service(item[:services_p])
+        new_facility.services = new_facility.services[0].to_s.gsub(',' , "")
+          .gsub(';' , "").gsub('vehicle' , "").gsub('VIN' , "").split
+        facilities << new_facility
+      elsif item[:state].include?("NY")
+        new_facility = Facility.new({
+          name: "#{item[:office_name]} #{item[:office_type]}", 
+          address: "#{item[:street_address_line_1]}, #{item[:street_address_line_2]} 
+          #{item[:city]}, #{item[:state]} #{item[:zip_code]}", 
+          phone: item[:public_phone_number],
+          hours: "Monday: #{item[:monday_beginning_hours]}-#{item[:monday_ending_hours]}; Tuesday: #{item[:tuesday_beginning_hours]}-#{item[:tuesday_ending_hours]}; Wednesday: #{item[:wednesday_beginning_hours]}-#{item[:wednesday_ending_hours]}; Thursday: #{item[:thursday_beginning_hours]}-#{item[:thursday_ending_hours]}; Friday: #{item[:friday_beginning_hours]}-#{item[:friday_ending_hours]}; Saturday: #{item[:saturday_beginning_hours]}-#{item[:saturday_ending_hours]}" })
+        facilities << new_facility
+      elsif item[:state].include?("MO")
+        new_facility = Facility.new({
+          name: item[:name], 
+          address: "#{item[:address1]}, #{item[:city]}, #{item[:state]} #{item[:zipcode]}", 
+          phone: item[:phone],
+          hours: item[:daysopen],
+          closed: item[:holidaysclosed]})
+        facilities << new_facility
+      end
+    end
+  end
```

Comment:
I think first and foremost, the `create_facilities` method is quite long. I would encourage you to keep your methods somewhere between 7 - 10 lines in length and refactor into helper methods to make the code more readable and easier to understand.

```ruby
def create_facilities(items)
  items.map do |item|
    state = item[:state]
    
    case state
    when "CO"
      create_co_facility(item)
    when "NY"
      create_ny_facility(item)
    when "MO"
      create_mo_facility(item)
    end
    
  end
end
```

I would also replace `items` with `dataset` or something similar, and `item` to `data` or something similar.

The last thing I want to mention is that I see that you are using a `map` but also shoveling the `new_facility` into a `facilities` array. This defeats the purpose of using a `map` and these lines can be removed since no shoveling needs to take place. A `map` will transform every element in the array from one thing to something else, which in this example are `Facility` objects.